crates/ty_ide/src/completion.rs

```diff
@@ -839,16 +846,22 @@ fn is_in_string(parsed: &ParsedModuleRef, offset: TextSize) -> bool {
     })
 }
 
-/// Order completions lexicographically, with these exceptions:
+/// Order completions according to the following rules:
 ///
-/// 1) A `_[^_]` prefix sorts last and
-/// 2) A `__` prefix sorts last except before (1)
+/// 1) Names with no underscore prefix
+/// 2) Names starting with `_` but not dunders
```

Comment:
Maybe it's out of scope for this PR, but perhaps deprecated items should also ranked  last? (and crossed out in the autocompletion list)

<img width="544" height="93" alt="Image" src="https://github.com/user-attachments/assets/44234814-e418-40bd-b2de-1ce8b7efde14" />